src/main/java/org/opensearch/ad/task/ADBatchTaskRunner.java

```diff
@@ -597,7 +597,8 @@ private void detectAnomaly(
                     error,
                     null,
                     adTask.getDetector().getUser(),
-                    anomalyDetectionIndices.getSchemaVersion(ADIndex.RESULT)
+                    anomalyDetectionIndices.getSchemaVersion(ADIndex.RESULT),
+                    null
```

Comment:
Why set to `null` here, can the model id not be pulled from the `rcf` variable? Where is the `model_id` actually added into the results exactly?

Reply:
^ I may have answered my own question - is this because this is related to historical detector results and `model_id` is not needed in that case? If so, how do we plan on handling the future case of historical HC detectors?

Reply:
yes, historical detector has no model id, bu task id.  Adding a model id may not be a good design as
* historical detector does not use it.  It uses task id. Front-end needs two sets of logic for real-time and historical detectors.
* it increases the size of result index doc.

We may be better off using the new result search API.  The only issue is the timeline.  We may not be able to finish appsec before 1.1 release.  Any suggestion?

Reply:
I see. Makes sense to not include it, and we can use `task_id` instead of `model_id` for aggregating HC historical results, if the new result search API is still not ready. Thanks for the clarification

Reply:
For historical results, we just store detector level task id in AD result index. I think we need to store model id in historical result as well to search entity's results.